auth/signaturevalidator.js

```diff
@@ -101,7 +101,9 @@ var limitClientWithExt = function(credentialName, issuingClientId, accessToken,
     // Validate certificate scopes are subset of client
     if (!utils.scopeMatch(scopes, [cert.scopes])) {
       throw new Error("ext.certificate issuer `" + issuingClientId +
-                      "` doesn't have sufficient scopes");
+                      "` doesn't satisfiy all certificate scopes " +
+                      cert.scopes.join(', ') + ".  The temporary " +
+                      "credentials were not generated correctly.");
```

Comment:
This is awesome...

One day we probably should use some template language to these errors, so that they can be multi-line, markdown, have be very explanatory.. Like the ones we recently added to tc-lib-api... And I mean for all the errors we report here, better messages would be good...

But this goes very very far to get things right...